src/core/server/plugins/plugin_context.ts

```diff
@@ -65,6 +66,19 @@ export function createPluginInitializerContext(
      * Core configuration functionality, enables fetching a subset of the config.
      */
     config: {
+      /**
+       * Global configuration
+       * Note: naming not final here, it will be renamed in a near future (https://github.com/elastic/kibana/issues/46240)
+       * @deprecated
+       */
+      globalConfig__deprecated$: coreContext.configService.getConfig$().pipe(
```

Comment:
I think we should only expose whitelisted fields. We don't want to expose the entire config tree to any plugin, only a handpicked list of configs that we know are needed by many plugins (I believe they should be listed in the issue).

I would expect this to look more like:

```ts
type SharedGlobalConfig = Readonly<{
  // Maybe use a recursive readonly type if we have one
  kibana: Readonly<{
   defaultAppId: string;
   index: string;
   disabledWelcomeScreen: string;
   // ...etc
 }>;
 path: Readonly<{
  data: string;
 }>;
 // ...etc
}

globalConfig__deprecated$: Observable<SharedGlobalConfig>;
```

Reply:
Oh! Thanks! I misunderstood the conversation in the original issue then. I'll rework this :)

Reply:
Yea, this is actually way better than my suggestion